packages/browser-lite/src/schema_validator.ts

```diff
@@ -0,0 +1,166 @@
+import * as ajv from 'ajv';
+
+import { JsonSchema } from './types';
+
+export interface SchemaValidationResult {
+    success: boolean;
+    errors: string[];
+    fatal?: string;
+}
+
+export interface SchemaValidator {
+    orderValidator: (input: string) => SchemaValidationResult;
+    messageValidator: (input: string) => SchemaValidationResult;
+}
+
+/**
+ * Constructs a `SchemaValidator` object that is compatible with the `orderfilter` package.
+ * @param chainId The chainId of the current ethereum network
+ * @param exchangeAddress The exchange address that Mesh should recognize on the current network
+ * @param customOrderFilter An extension of the order schema that adds specific requirements to
+ *        orders.
+ */
+export function getSchemaValidator(
+    chainId: number,
+    exchangeAddress: string,
+    customOrderFilter?: JsonSchema,
+): SchemaValidator {
+    const chainIdSchema = {
+        $async: false,
+        $id: '/chainId',
+        const: chainId,
+    };
+
+    // NOTE(jalextowle): The current implementation of `zeroex.SignedOrder.MarhsalJSON`
+    // lowercases all addresses. This eliminates the need to add the checksummed address.
+    const exchangeAddressSchema = {
+        $async: false,
+        $id: '/exchangeAddress',
+        enum: [exchangeAddress],
+    };
+
+    const AJV = new ajv();
+    AJV.addSchema(addressSchema);
+    AJV.addSchema(wholeNumberSchema);
+    AJV.addSchema(hexSchema);
+    AJV.addSchema(chainIdSchema);
+    AJV.addSchema(exchangeAddressSchema);
+    AJV.addSchema({
+        ...customOrderFilter,
+        $async: false,
+        $id: '/customOrder',
+    });
+    AJV.addSchema(orderSchema);
+    AJV.addSchema(signedOrderSchema);
+    AJV.addSchema(rootOrderSchema);
+    AJV.addSchema(rootOrderMessageSchema);
+
+    const orderValidate = AJV.getSchema('/rootOrder');
+    if (orderValidate === undefined) {
+        throw new Error('Cannot find "/rootOrder" schema in AJV');
+    }
+    const messageValidate = AJV.getSchema('/rootOrderMessage');
+    if (messageValidate === undefined) {
+        throw new Error('Cannot find "rootOrderMessage" schema in AJV');
+    }
+    return {
+        orderValidator: constructValidationFunctionWrapper(orderValidate),
+        messageValidator: constructValidationFunctionWrapper(messageValidate),
+    };
+}
+
+function constructValidationFunctionWrapper(
+    validationFunction: ajv.ValidateFunction,
+): (input: string) => SchemaValidationResult {
+    return (input: string) => {
+        const result: any = { success: false, errors: [] };
+        try {
+            result.success = validationFunction(JSON.parse(input));
+            if (validationFunction.errors) {
+                result.errors = validationFunction.errors.map(error => JSON.stringify(error));
+            }
+        } catch (error) {
+            result.fatal = JSON.stringify(error);
+        }
+        return result;
+    };
+}
+
+const addressSchema = {
+    $async: false,
```

Comment:
I think we can remove `$async: false` everywhere now. AFACT from looking at the `ajv` docs, it doesn't actually do anything unless you call async methods like `compileAsync`, which we are not.

Eventually, I would also like to use one single canonical source for our JSON Schemas to eliminate potential bugs/headaches down the road. For example, we could use the Go definitions and pass them through to `ajv` (e.g. by calling `ajv.addSchema` from Go). But if you like we can do that in a separate PR.